lib/os/onoff.c

```diff
@@ -142,6 +144,15 @@ static void onoff_start_notify(struct onoff_service *srv,
 	k_spinlock_key_t key = k_spin_lock(&srv->lock);
 	sys_slist_t clients = srv->clients;
 
+	/* Cancellation occurred, start notification is irrelevant */
+	if ((srv->flags & SERVICE_STATE_MASK) == SERVICE_STATE_TO_OFF) {
```

Comment:
Protection against case when `onoff_cancel` was called and transition to off initiated but before cancel function is called transition to on occurs. In that case, it shall be skipped because service will be cancelled and notification from cancellation will come to the user.